### PR TITLE
CNV-1403 containerdisk source

### PR DESCRIPTION
**Pull request by aburdenthehand**
Branched off #25359 as the concept and 'preparing' procedure serves as the foundation for this. The rest of this is the same as the 'Importing a VM image into a DataVolume'  conditionalised so that 'container disk' is used instead of 'virtual machine image', and so that the the config `spec:source`, registry example, and prereqs suit container disks.

Build: https://cnv-1403-containerdisk-source--ocpdocs.netlify.app/openshift-enterprise/latest/virt/virtual_machines/importing_vms/virt-importing-container-disk-with-datavolumes.html

https://issues.redhat.com/browse/CNV-1403
https://issues.redhat.com/browse/CNV-1398

### CODE REVIEW

File: modules/virt-preparing-container-disk-for-vms.adoc
Comment (dankenigsberg) [inline modules/virt-preparing-container-disk-for-vms.adoc:24]: before that, we need to ensure that the image would be readable by qemu
```
chmod 440 <vm_image>.qcow2
```

File: modules/virt-preparing-container-disk-for-vms.adoc
Comment (dankenigsberg) [inline modules/virt-preparing-container-disk-for-vms.adoc:49]: ditto

**Review by dankenigsberg — CHANGES_REQUESTED** [review]
I like this addition, and think that customers would value the added warning against superfluous usage of containerDisks.

File: _topic_map.yml
Comment (davidvossel) [inline _topic_map.yml:2314]: I'm not sure why this should have it's own section rather than being rolled into the `Importing virtual machine images with DataVolumes` that already exists. 

This is really similar to the existing http import documentation, The only difference is that we're using the DataVolume "registery" source instead of "http"

File: modules/virt-about-container-disks.adoc
Comment (davidvossel) [inline modules/virt-about-container-disks.adoc:11]: The import destination is a PVC, not a DataVolume. 

The DataVolume itself represents the automated workflow of importing an image source "s3, http, registry, etc..." into a PVC. So when we talk about importing, i think the emphasis should be on a DataVolume importing a containerDisk onto a PVC for persistence rather than "importing into a DataVolume"

File: modules/virt-about-container-disks.adoc
Comment (davidvossel) [inline modules/virt-about-container-disks.adoc:13]: something like "Importing container disks into PVCs using DataVolumes" might be more accurate

File: modules/virt-about-container-disks.adoc
Comment (davidvossel) [inline modules/virt-about-container-disks.adoc:15]: like the other comments, perhaps reword to indicate "import into pvc"

File: modules/virt-preparing-container-disk-for-vms.adoc
Comment (davidvossel) [reply modules/virt-preparing-container-disk-for-vms.adoc:24]: our test suite currently verifies with 444, not 440.

440 might work, but that isn't tested at the moment.

File: modules/virt-importing-vm-datavolume.adoc
Comment (davidvossel) [inline modules/virt-importing-vm-datavolume.adoc:104]: Can we get an example here that works if some attempts to copy and paste it? 

For example, ```"docker://kubevirt/fedora-cloud-container-disk-demo:latest"``` is one of our upstream images that would actually work on their cluster.

File: modules/virt-preparing-container-disk-for-vms.adoc
Comment (davidvossel) [reply modules/virt-preparing-container-disk-for-vms.adoc:24]: I added a PR (https://github.com/kubevirt/kubevirt/pull/4416) to KubeVirt to both build and test our containerDisks using 107:107 ownership and 440 file mode instead of the previous 444 mode.

So, once we get this build/test change merged, our officially tested recommendation can be 440.

File: modules/virt-preparing-container-disk-for-vms.adoc
Comment (davidvossel) [inline modules/virt-preparing-container-disk-for-vms.adoc:34]: In order to set the 440 file mode, we can use a multi-stage docker build. 

here's an example that uses the standard ubi container image as the builder image, then uses a scratch image to store the result. 

```
FROM registry.access.redhat.com/ubi8/ubi:latest AS builder
ADD --chown=107:107 Fedora-Cloud-Base-32-1.6.x86_64.qcow2 /disk/
RUN chmod 0440 /disk/*

FROM scratch
COPY --from=builder /disk/* /disk/
```

File: _topic_map.yml
Comment (aburdenthehand) [reply _topic_map.yml:2314]: You make a really good point. That was my first thought but looking at the preparation steps and concepts required for this I thought it might be best to have a complete user story for container disks - what they are, how to prep them, specific examples, etc - and to improve searchability. However I've followed up with a few people and have learnt that searchability won't be impacted, and I think I can now see a way of handling this without having too many messy optional/deviations, like "If you've got X you need to have done Y but if you've got Z then you must A." 
I'll fix up the other feedback (which will go into #25359) so I can get that PR merged and then come back to reorganising this content. We're getting close to a docs freeze but I'll try and make this work before that happens next week.

File: modules/virt-importing-vm-datavolume.adoc
Comment (aburdenthehand) [reply modules/virt-importing-vm-datavolume.adoc:104]: registry.fedoraproject.org/fedora works for me as an image but if the kubevirt-maintained image is a better source I can update the example
I can see that I've messed the conditionals here though.

File: modules/virt-preparing-container-disk-for-vms.adoc
Comment (aburdenthehand) [reply modules/virt-preparing-container-disk-for-vms.adoc:24]: @davidvossel Have updated the Dockerfile step and explanation based on the comments here and in #25359. Can you please confirm this is correct now?

File: modules/virt-importing-vm-datavolume.adoc
Comment (davidvossel) [reply modules/virt-importing-vm-datavolume.adoc]: the issue is that ```registry.fedoraproject.org/fedora works``` doesn't have a disk image in it. It's a generic container image. We're actually placing a qcow disk in the ```kubevirt/fedora-cloud-container-disk-demo:latest``` image

Comment (aburdenthehand) [reply modules/virt-importing-vm-datavolume.adoc]: Ahh, understood. Thanks for clarifying.

File: modules/virt-preparing-container-disk-for-vms.adoc
Comment (aburdenthehand) [reply modules/virt-preparing-container-disk-for-vms.adoc:24]: Followup: I got ack from David for the Dockerfile update and 25359 is now merged.

**Comment by aburdenthehand** [comment]
Closing this in favour of #27060, which incorporates importing a container disk into the existing importing a VM image into a DataVolume procedure and is neater

**Pull request closed without merging**
